drivers/copy/copy_test.go

```diff
@@ -0,0 +1,179 @@
+// +build linux
+
+package copy
+
+import (
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/containers/storage/pkg/parsers/kernel"
+	"github.com/containers/storage/pkg/system"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sys/unix"
+)
+
+func TestIsCopyFileRangeSyscallAvailable(t *testing.T) {
+	// Verifies:
+	// 1. That copyFileRangeEnabled is being set to true when copy_file_range syscall is available
+	// 2. That isCopyFileRangeSyscallAvailable() works on "new" kernels
+	v, err := kernel.GetKernelVersion()
+	require.NoError(t, err)
+
+	copyWithFileRange := true
+	copyWithFileClone := false
+	doCopyTest(t, &copyWithFileRange, &copyWithFileClone)
+
+	if kernel.CompareKernelVersion(*v, kernel.VersionInfo{Kernel: 4, Major: 5, Minor: 0}) < 0 {
+		assert.False(t, copyWithFileRange)
+	} else {
+		assert.True(t, copyWithFileRange)
+	}
```

Comment:
Since I expect we're going to want this feature to be used when the `FICLONE` ioctl has been backported, can we make this test key off of something other than the reported kernel version?

Reply:
Any idea how?

Reply:
@rhvgoyal Any suggestions?

Reply:
This check has just been removed in https://github.com/moby/moby/pull/36736

Reply:
@redbaron Thanks for pointing this out, I have removed it from this PR also.